Ishaym/kafka exercise section 4 1

Ishai-Marom opened the pull request:
Kafka Exercise - Ishai Marom
    
section 4.1, setting up producer and consumer of strings.
recent change: adding a log4j-simple dependency to print `info` level
logs.

Ishai-Marom committed:
Kafka Exercise - Ishai Marom

section 4.1, setting up producer and consumer of strings

Ishai-Marom committed:
Kafka Exercise - Ishai Marom

section 4.1, setting up producer and consumer of strings.
recent change: adding a log4j-simple dependency to print `info` level
logs.

Ishai-Marom committed:
Ishai Marom - Kafka Exercise 4.1

fixed according to the review

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Consumer/src/main/java/org/ishaym/training/KafkaEnvironmentSetUp.java:
```diff
@@ -0,0 +1,79 @@
+    private static final String CLIENT_ID = "consumer-client-id";
+    private static final int NUMBER_OF_PARTITIONS = 1;
+    private static final int FACTOR_OF_REPLICAS = 1;
+    private static final int TIMEOUT = 10;
```
Extract to constants file

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Consumer/src/main/java/org/ishaym/training/ExerciseKafkaConsumer.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    public void consume() {
+        Thread thread = new Thread(
```
It is better to make the class to implement Runnable in order to make the use of different thread more clear and easy

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Consumer/src/main/java/org/ishaym/training/common/Common.java:
```diff
@@ -0,0 +1,33 @@
+import java.io.IOException;
+import java.util.Objects;
+
+public class Common {
```
Rename to Constants

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Consumer/src/main/java/org/ishaym/training/ExerciseKafkaConsumer.java:
```diff
@@ -0,0 +1,65 @@
+import java.util.Collections;
+import java.util.Properties;
+
+public class ExerciseKafkaConsumer {
```
No need for Exercise prefix

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Consumer/src/main/java/org/ishaym/training/KafkaEnvironmentSetUp.java:
```diff
@@ -0,0 +1,79 @@
+            TimeoutException {
+        LOGGER.debug("started creating the consumer environment");
+
+        String topic = Common.STRING_MESSAGES_TOPIC;
```
There is no need to make assignment on this case, just use the constant value

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Consumer/src/main/java/org/ishaym/training/Main.java:
```diff
@@ -0,0 +1,34 @@
+        try {
+            KafkaEnvironmentSetUp.setUp();
+            ExerciseKafkaConsumer consumer = new ExerciseKafkaConsumer();
+            consumer.subscribe(Common.STRING_MESSAGES_TOPIC);
```
You can place this line inside the constructor

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Producer/src/main/java/org/ishaym/training/ExerciseKafkaProducer.java:
```diff
@@ -0,0 +1,47 @@
+import java.io.IOException;
+import java.util.Properties;
+
+public class ExerciseKafkaProducer {
```
There is no need for Exercise prefix

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Producer/src/main/java/org/ishaym/training/KafkaEnvironmentSetUp.java:
```diff
@@ -0,0 +1,77 @@
+public class KafkaEnvironmentSetUp {
+    private static final Logger LOGGER = LogManager.getLogger(KafkaEnvironmentSetUp.class);
+
+    private static final String CLIENT_ID = "producer-client-id";
+    private static final int NUMBER_OF_PARTITIONS = 1;
+    private static final int FACTOR_OF_REPLICAS = 1;
+    private static final int TIMEOUT = 10;
```
Extract to constants file

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Producer/src/main/java/org/ishaym/training/KafkaEnvironmentSetUp.java:
```diff
@@ -0,0 +1,77 @@
+        return adminClient.listTopics().names().get().contains(topic);
+    }
+
+    private static void createTopic(AdminClient adminClient, String topicName)
```
You need to create the topic only on the consumer side, on the producer side you should wait until the topic has been created in order to ensure that the consumer side is on and ready to receive messages

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Producer/src/main/java/org/ishaym/training/KafkaEnvironmentSetUp.java:
```diff
@@ -0,0 +1,77 @@
+            TimeoutException {
+        LOGGER.debug("started creating the producer environment");
+
+        String topic = Common.STRING_MESSAGES_TOPIC;
```
Same as above

asafbena commented on Exercises/KafkaExercise/Exercise_4_1/Producer/src/main/java/org/ishaym/training/common/Common.java:
```diff
@@ -0,0 +1,33 @@
+import java.io.IOException;
+import java.util.Objects;
+
+public class Common {
```
Same as above

Ishai-Marom committed:
Ishai Marom - Kafka Exercise section 4.1

section 4.1, fixed according to the second review.

Ishai-Marom committed:
Ishai Marom - Kafka Exercise Section 4.1

section 4.1, fixed according to review on section 4.2.
in addition, changed the config classes to the auto-generated classes
from plugin that generates jackson classes from json.

Ishai-Marom committed:
Ishai Marom - Kafka Exercise Section 4.1

section 4.1, fixed according to review on section 4.2.

Ishai-Marom committed:
Ishai Marom - Kafka Exercise Section 4.1

section 4.1, fixed according to review on section 4.3.

Ishai-Marom committed:
Ishai Marom - Kafka Exercise Section 4.1

section 4.1, fixed according to review on section 4.3.
